### ISSUE TITLE
Reporting nonexistent columns

### ISSUE THREAD

**Issue by KamasamaK** [cflint/CFLint#522]
Using CFLint-1.3.0

Given the following component named `test.cfc`:

```
/**
 * Test hint
 */
component {
	/**
	 * Converts date to formatted string-representation of date
	 * @dateObj The date object to format
	 */
	public string function formatDate(
		required date dateObj
	)
	{
		return dateFormat(arguments.dateObj, "mm/dd/yyyy");
	}
}
```

I get the following issues:

1. ARGUMENT_HAS_PREFIX_OR_POSTFIX (9, 53)
1. ARGUMENT_IS_TEMPORARY (9, 53)
1. COMPONENT_INVALID_NAME (1, 24)

None of those columns exist on those respective lines.

**Comment by ryaneberly** [comment]
@KamasamaK  I believe this is fixed in https://github.com/cflint/CFLint/tree/522_colu

**Comment by ryaneberly** [comment]
It seems to be a 0 vs 1 based index issue.  I'm not entirely happy with the current patch/fix.

**Comment by KamasamaK** [comment]
I believe I asked you at one point which was being used and you said it was 1-based. Looking at some of the issues again recently, it appeared to me that Line was 1-based and Column was 0-based though I'm not sure if that is universally true. For my use, I think #491 would be preferable anyway.

If that's the root cause, it wasn't apparent since the numbers were not off by 1, but off by a lot.

**Comment by KamasamaK** [comment]
After testing on that branch, I think I see what you mean. After the patch, it is reporting ARGUMENT_HAS_PREFIX_OR_POSTFIX and ARGUMENT_IS_TEMPORARY as still being on line 9 instead of 10, but correctly reporting column 11 for the argument assuming 0-based index. Is that a 0- vs 1-based issue? Adding another argument on the following line results in the same issue of reporting on that original line but using a nonexistent column. Take the following example:

```
/**
 * Test hint
 */
component {
	/**
	 * Converts date to formatted string-representation of date
	 * @dateObj The date object to format
	 * @formatStr The format string mask
	 */
	public string function formatDate(
		required date dateObj,
		string formatStr="mm/dd/yyyy"
	)
	{
		return dateFormat(arguments.dateObj, arguments.formatStr);
	}
}
```

It is now reporting

1. ARGUMENT_HAS_PREFIX_OR_POSTFIX (10, 16)
2. ARGUMENT_HAS_PREFIX_OR_POSTFIX (10, 35)
3. ARGUMENT_IS_TEMPORARY (10, 16)
4. COMPONENT_INVALID_NAME (1, 19)


This is ignoring some VAR_HAS_PREFIX_OR_POSTFIX and VAR_IS_TEMPORARY reports, which are being reported on the argument usage on line 15. You can see **2** and **4** are still reporting nonexistent columns.

**Comment by ryaneberly** [comment]
merged to dev, assigned to 1.4.0

**Comment by ryaneberly** [comment]
getting line and column right is doable, but is going to take some rework.  Moving this out to 1.4.1.  

I added #491 back to 1.4.0.
